src/analyses/mCP.ml

```diff
@@ -65,6 +65,12 @@ struct
     let deps (x,_) = iter (check_dep x) @@ (find_spec x).dep in
     iter deps xs
 
+  let specific_cont_sens xs =
+    (* most analysis must be set to context-insensitive, because we only want to analyse context-sensitive for one specific analysis *)
+    let sens_ana = ["callstring_fundec"; "callstring_stmt"; "callstring_loc"; "loopfree_callstring"] in
+    let enabled = List.fold_left (fun acc x -> acc || (mem x xs)) false sens_ana in
+    (* returns the edited list of insensitive analyses if enabled *)
+    if enabled then Some(filter (fun x -> not (mem x sens_ana)) xs) else None
```

Comment:
I don't think we need such magic behavior where activating some analysis will quietly change our usual context-sensitivity configuration. If the user wants only callstring contexts without anything else, then they can configure that with existing options.

Reply:
>If the user wants only callstring contexts without anything else, then they can configure that with existing options.

This is true, however, it is not really user-friendly to configure, since one has to add all the other analyses into the set of context-insensitive analyses. This is even more cumbersome to do when one also uses the auto-tuner, which may itself activate some analyses.

I think there should at least be an easy option for a user to let the implementation behave as here.

Reply:
I agree `ana.ctx_insens` is a bit backwards and inconvenient for this. Maybe we could have two complementary options:
1. `ana.ctx_insens` (like a blacklist) as it is right now.
2. `ana.ctx_sens`/`ana.ctx_only` (like a whitelist) for a list of _only_ analyses that should be context-sensitive.

Both cannot be used at the same time, so the logic could be that if `ana.ctx_sens` is empty (as by default), then `ana.ctx_insens` is used, otherwise only `ana.ctx_sens` is considered.

Reply:
I am currently at the realization of this, but with those two options it's not possible to configure a context-insensitive analysis easily. Because if I set the `ana.ctx_sens` to an empty array, the `ana.ctx_insens` analysis is used and would need to contain all activated analyses. Currently it's possible to make the analysis context-insensitive with the context gas analysis and a gas of 0. 
Should I include an separate option or a string for the array (for example `all` in the `ana.ctx_insens` analysis to set all activated analyses insensitive), which allows it to make the analysis context-insensitive?

Reply:
That's a good point. I would say it's fine because we never had a way to make a fully context-insensitive analysis anyway (other than adding everything to `ana.ctx_insens`, which will still work) and context gas 0 is another way.

In the future, we could make the schema more flexible (e.g. allow `null` values for a special meaning) or use `anyOf`s, if we really want a clean way to do that.

For a fully context-insensitive analysis, one could even imagine defining an optimized constraint system whose local unknowns are just CFG nodes, not pairs where the second context component will be effectively degraded to unit.